examples/preprocessing/plot_target_encoder.py

```diff
@@ -0,0 +1,224 @@
+# categorical feature in gradient boosting.
+high_cardinality_features = n_unique_categories[n_unique_categories > 255].index
+low_cardinality_features = n_unique_categories[n_unique_categories <= 255].index
+mixed_encoded_preprocessor = ColumnTransformer(
+    [
+        ("numerical", "passthrough", numerical_features),
+        (
+            "high_cardinality",
+            TargetEncoder(target_type="continuous"),
+            high_cardinality_features,
+        ),
+        (
+            "low_cardinality",
+            OrdinalEncoder(handle_unknown="use_encoded_value", unknown_value=-1),
+            low_cardinality_features,
+        ),
+    ],
+    verbose_feature_names_out=False,
+)
+
+# The output of the of the preprocessor must be set to pandas so the
+# gradient boosting model can detect the low cardinality features.
+mixed_encoded_preprocessor.set_output(transform="pandas")
+mixed_pipe = make_pipeline(
+    mixed_encoded_preprocessor,
+    HistGradientBoostingRegressor(
+        random_state=0, max_iter=max_iter, categorical_features=low_cardinality_features
+    ),
+)
+mixed_pipe
+
+# %%
+# Finally, we evaluate the pipeline using cross validation and record the results:
+evaluate_model_and_store("mixed_target", mixed_pipe)
+
+# %%
+# Plotting the Results
+# --------------------
+# In this section, we display the results for all the encoders. When evaluating the
+# predictive performance on the test set, dropping the categories perform the
+# worst and the target encoders performs the best.
+# - Dropping the categorical features makes the pipeline less expressive and
+#   underfitting as a result;
+# - Due to the high cardinality and to reduce the training time, the one-hot
+#   encoding scheme uses `max_categories=20` which prevents the features from
+#   expanding too much, which can result in underfitting.
+# - The one-hot encoding scheme makes the pipeline overfitting as the larger
+#   number of features explodes with rare category occurrences that are correlated
+#   with the target by chance (on the training set only);
+# - The ordinal encoding imposes an arbitrary order to the features which are then
+#   treated as numerical values by the
+#   :class:`~sklearn.ensemble.HistGradientBoostingRegressor`. Since this
+#   model groups numerical features in 256 bins per feature, many unrelated categories
+#   can be grouped together and as a result overall pipeline can underfit;
+# - When using the target encoder, the same binning happens, but since the encoded
+#   values are statistically ordered by marginal association with the target variable,
+#   the binning use by the :class:`~sklearn.ensemble.HistGradientBoostingRegressor`
+#   makes sense and leads to good results: the combination of smoothed target
+#   encoding and binning works as a good regularizing strategy against overfitting while
+#   not limiting the expressiveness of the pipeline too much.
```

Comment:
I would rather move the analysis below the plot, to avoid having to scroll through the code of the plot when reading the analysis.